internal/api/v3.go

```diff
@@ -88,6 +89,11 @@ func MakeV3Courses(d *data.Data) map[string]*V3Course {
       continue
     }
 
+    code := strings.Join(strings.Split(id, " ")[:2], " ")
```

Comment:
A quick skim through the sample `coursesection_1.csv` shows that this may not be robust on IDs like `WRIT165IOSC-01 SP2021`.  Since you're resorting to regexp, we might as well regexp the whole thing:

```go
// define at top level
var reCourseCode = regexp.MustCompile(`^([A-Z]+) *([0-9A-Z]+) *([A-Z]{2})-(\d{2}) ([A-Z0-9+)$`)

// within function
matches, err := reCourseCode.FindStringSubmatch(id)
if err != nil {
  // catch some error or something
}
// matches[1] is department, matches[2] is code, matches[3] is section, etc.
// use `fmt.Sprintf("%s %s")` to do joining instead of +
```

Reply:
replaced with regex